account_invoice_refund_link/models/account_invoice.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2004-2011 Pexego Sistemas Informáticos. (http://pexego.es)
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# Copyright 2014-2018 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    refund_invoice_ids = fields.One2many(
+        "account.move", "reversed_entry_id", string="Refund Invoices", readonly=True
+    )
+
+    @api.model
+    def _reverse_move_vals(self, default_values, cancel=True):
+        move_vals = super()._reverse_move_vals(default_values, cancel)
+        if move_vals["type"] in ("out_refund", "in_refund"):
+            refund_lines_vals = move_vals.get("line_ids", [])
+            for i, line in enumerate(self.line_ids):
```

Comment:
```suggestion
            for i, line in enumerate(self.invoice_line_ids):
```

Reply:
move_vals doesn't have invoice_line_ids key. I could search for each invoice_line_ids which line_ids corresponds to it but it implie an extra cycle. I think I can put a condition to bind with origin_line_ids only if the line has the option exclude_from_invoice_tab false. Check the change I made and tell me what you think

Reply:
good :+1: